Migrating MBI KB Content to DevDocs (#8)
magento/devdocs-mbi: docs/export-api.md

---
group: mbi
title: Automating Data Retrieval with the Data Export API
zendesk_id: 360016732271
functional_areas:
  - Export API
  - System
  - Setup
---

> The Magento BI API is provided to customers who wish to access some of the raw data behind their Magento BI dashboard.

Reviewer: Make this paragraph more active voice. Perhaps the first sentence could be

The Magento BI API allows customers to access...

> The API is entirely HTTP-based. Methods that retrieve data must be accessed with a GET request, and those that create, modify or delete data with a POST request.

Reviewer: ```suggestion
The API is entirely HTTP-based. Use a GET request to retrieve data or a POST request to create, modify, or delete data.
```

### Authentication

> API connections are authenticated via a token that must be passed in the HTTP header of every request.

Reviewer: ```suggestion
API connections are authenticated using a token that must be passed in the HTTP header of every request.
```

> 1. Keys can be generated via **Export Data -> Export API**, and must be passed in the "X-RJM-API-Key" portion of every request header.

Reviewer: ```suggestion
1. Generate a key from  **Export Data -> Export API**. You must specify this key in the "X-RJM-API-Key" header of every request.
```

Reviewer: Also, these steps should be bullet items.

> 1. All keys must be associated with a single client and a set of IP addresses that the requests will be made from.  IP addresses can be specified either as a specific address, or as a range of addresses in [CIDR notation](http://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing).  For example, the CIDR notation to allow ALL IP addresses would be: "0.0.0.0/0".

Reviewer: ```suggestion
1. All keys must be associated with a single client and a set of IP addresses that are allowed to make requests.  IP addresses can be specified either as a specific address, or as a range of addresses in [CIDR notation](http://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing).  For example, the CIDR notation to allow ALL IP addresses would be: "0.0.0.0/0".
```

---
group: mbi
title: Magento BI Export API
zendesk_id: 360016732271
functional_areas:
  - Export API
  - System
  - Setup
---

The Magento BI API allows customers to access some of the raw data behind their Magento BI dashboard. The API is entirely HTTP-based.

## Authentication

API connections are authenticated using a token that must be passed in the HTTP header of every request.

* Generate a key from **Export Data -> Export API**. You must specify this key in the `X-RJM-API-Key` header of every request.

* All keys must be associated with a single client and a set of IP addresses that are allowed to make requests. IP addresses can be specified either as a specific address, or as a range of addresses in [CIDR notation](http://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing). For example, the CIDR notation to allow ALL IP addresses would be: `0.0.0.0/0`.

![mbi-valid-ip-address.png](../docs/images/mbi-valid-ip-address.png)

> ## Methods

Reviewer: ```suggestion
## HTTP Methods
```

Use a GET request to retrieve data or a POST request to create, modify, or delete data.

Methods are defined by an HTTP Verb (GET, POST, PUT, DELETE), a URL and a set of arguments. The response HTTP status code will be 200-Success for all successful requests unless otherwise specified.

> ### Exporting Raw Data Exports

Reviewer: Do all the MBI headings use init caps for all significant words?

Author: I'm sure we're inconsistent throughout. Will address in a future clean up.

{:.bs-callout-info}
You need a [raw data export](https://docs.magento.com/mbi/tutorials/export-raw-data.html) before calling the following methods.

**Endpoint**

> `GET /export \[no arguments\]`

Reviewer: ```suggestion
`GET /export`
```

---
group: mbi
title: Magento BI Export API
zendesk_id: 360016732271
functional_areas:
  - Export API
  - System
  - Setup
---
​

> The Magento BI Export API allows you to  to access some of the raw data behind your Magento BI dashboard.

Reviewer 2: double "to" in line 11

> Before using the import API, make sure you [authenticate](../docs/getting-started.md#authentication) your connection.

Reviewer 2: "import" should read "export"
